src/index.ts

```diff
@@ -0,0 +1,154 @@
+/**
+ * @fileoverview This is an ES Module which exports `vizObserver`, see its function below.
+ *
+ * This uses a combination of IntersectionObserver and ResizeObserver to inform you when DOM
+ * nodes move around on a page (and also when they're resized). This might be useful if
+ * you're trying to track a complex HTML structure, perhaps one you don't control.
+ *
+ * It also works with target nodes inside Shadow DOM.
+ */
+
+/**
+ * This is an element's position relative to the page's documentElement.
+ */
+export interface Rect {
+  appear: boolean;
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+}
+
+export interface Options {
+  signal?: AbortSignal;
+}
+
+// const debug = false;
+const root = document.documentElement;
+
+const handlers = new WeakMap<Element, () => void>();
+
+const activeObservers = new Set<() => void>();
+handlers.set(root, () => activeObservers.forEach((handler) => handler()));
+
+const ro = new ResizeObserver((entries) => {
+  for (const entry of entries) {
+    handlers.get(entry.target)?.();
+  }
+});
+
+/**
+ * Sets up an observer that notifies on element resize or move.
+ *
+ * Returns a method to remove the observer (or accepts an AbortSignal). You should be sure to clean
+ * up outstanding observers even if you otherwise discard the element, otherwise you could leak
+ * memory.
+ */
+export default function vizObserver(element: Element, callback: (rect: Rect) => void) {
```

Comment:
I'm also wondering if we should make this a class like `ResizeObserver` instead of a callback. Each time `vizObserver()` is called we need to create new closures at scale this could be expensive for memory and collecting that memory. With a class a closure like `refresh` will only be instantiated once in total instead of once per usage.

Reply:
Yeah, idiomatically I like the idea of matching those other APIs.

Reply:
Should we call it `MovementObserver`?

Reply:
> I mean, the repo is called `VizObsever` ;)
> 
> My 2c is that `MovementObserver` doesn't imply that it also handles
resizing, which this approach does (albeit by wrapping `ResizeObserver`). I'm aware this is bikeshedding. Yet somethng like `ChangeObserver` is too generic.

Fair points! Lets go with `VizObserver` then :)

Reply:
Oh what about `VisualObserver`?